convert2rhel/backup/files.py

```diff
@@ -0,0 +1,146 @@
+        if not self.enabled or not os.path.isfile(backup_filepath):
+            loggerinst.info("%s hasn't been backed up." % self.filepath)
+            return
+
+        try:
+            shutil.copy2(backup_filepath, self.filepath)
+        except (OSError, IOError) as err:
+            # Do not call 'critical' which would halt the program. We are in
+            # a rollback phase now and we want to rollback as much as possible.
+            # IOError for py2 and OSError for py3
+            loggerinst.warning("Error(%s): %s" % (err.errno, err.strerror))
+            return
+
+        if rollback:
+            loggerinst.info("File %s restored." % self.filepath)
+            super(RestorableFile, self).restore()
+        else:
+            loggerinst.debug("File %s restored." % self.filepath)
+            # not setting enabled to false since this is not being rollback
+            # restoring the backed up file for conversion purposes
+
+    # Probably will be deprecated and unusable since using the BackupController
+    # Depends on specific usage of this
+    def remove(self):
+        """Remove restored file from original place, backup isn't removed"""
+        try:
+            os.remove(self.filepath)
+            loggerinst.debug("File %s removed." % self.filepath)
+        except (OSError, IOError):
+            loggerinst.debug("Couldn't remove restored file %s" % self.filepath)
+
+
+class MissingFile(RestorableChange):
+    """
+    File not present before conversion. Could be created during
+    conversion so should be removed in rollback.
+    """
+
+    def __init__(self, filepath):
+        super(MissingFile, self).__init__()
+        self.filepath = filepath
+
+    def enable(self):
+        if self.enabled:
+            return
+
+        if os.path.isfile(self.filepath):
+            loggerinst.debug(
+                "Shouldn't be called, file {filepath} is present before conversion".format(filepath=self.filepath)
+            )
+            return
+
+        loggerinst.info("Marking file {filepath} as missing on system.".format(filepath=self.filepath))
+        super(MissingFile, self).enable()
+
+    def restore(self):
+        if not self.enabled:
+            return
+
+        loggerinst.task("Rollback: remove file created during conversion {filepath}".format(filepath=self.filepath))
+
+        if not os.path.isfile(self.filepath):
+            loggerinst.info("File {filepath} wasn't created during conversion".format(filepath=self.filepath))
+        else:
+            try:
+                os.remove(self.filepath)
+                loggerinst.info("File {filepath} removed".format(filepath=self.filepath))
+            except OSError as err:
+                # Do not call 'critical' which would halt the program. We are in
+                # a rollback phase now and we want to rollback as much as possible.
+                loggerinst.warning("Error(%s): %s" % (err.errno, err.strerror))
+                return
```

Comment:
Same here, do we want to use `critical_no_exit()?`

Reply:
Maybe it would help if we made the choices `critical` and `critical_exit`? The one that exits is the one with the side effect (and a significant side effect at that), which argues that it should be the one with the more detailed name. It would also make the code more clear to someone just browsing it.

Reply:
Agreed @jochapma, we made it this way for making it easier to migrate later. Easier to add `critical_no_exit()` and call it than to refactor the whole codebase. Technical debt in the making but something we had to do last year